### PR TITLE
[Compute] `az vm host`: Add commands to support getting possible resize options and resizing dedicated host

### PR DESCRIPTION
**Pull request by yanzhudd**
**Related command**

`az vm host resize`
`az vm host list-host-resize-options`

**Description**

A new feature called host resize is added. The Customers will make az vm host list-host-resize-options to get possible resize options and resize their dedicated host to.
Close: #25999

**Testing Guide**

**History Notes**

[Compute] `az vm host resize`: Add new command to support resizing dedicated host
[Compute] `az vm host list-resize-options`: Add new command to support getting possible resize options

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/aaz/latest/vm/host/_list_host_resize_options.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+# --------------------------------------------------------------------------------------------
+
+# pylint: skip-file
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "vm host list-host-resize-options",
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/aaz/latest/vm/host/_list_host_resize_options.py:15]:
```suggestion
    "vm host list-resize-options",
```

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/vm/aaz/latest/vm/host/_list_host_resize_options.py:15]: @cakarata I suggest changing the command name from `vm host list-host-resize-options` to `vm host list-resize-options`, which is more concise and makes it easier for customers to input, what do you think of it?

Comment (cakarata) [reply src/azure-cli/azure/cli/command_modules/vm/aaz/latest/vm/host/_list_host_resize_options.py:15]: > 

I think it makes sense. I wanted to make it consistent with vm resize command but "vm host list-host-resize-options" is pure ugly.

Comment (yanzhudd) [reply src/azure-cli/azure/cli/command_modules/vm/aaz/latest/vm/host/_list_host_resize_options.py]: okay, I will rename this command.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**